rotkehlchen/chain/ethereum/transactions.py

```diff
@@ -52,41 +73,57 @@ def query_for_graph_delegation_txns(self, addresses: Sequence[ChecksumEvmAddress
                 location=Location.ETHEREUM,
                 event_types=[HistoryEventType.INFORMATIONAL],
                 event_subtypes=[HistoryEventSubType.APPROVE],
+                location_labels=[address],
             )
             dbevents = DBHistoryEvents(self.database)
             with dbevents.db.conn.read_ctx() as cursor:
-                if len(dbevents.get_history_events(
-                        cursor=cursor,
-                        filter_query=db_filter,
-                        has_premium=True,
-                )) == 0:
+                events = dbevents.get_history_events(
+                    cursor=cursor,
+                    filter_query=db_filter,
+                    has_premium=True,
+                )
+                if len(events) == 0:
                     continue
 
-            with self.database.conn.read_ctx() as cursor:
-                from_block = staking_contract.deployed_block
+                # find the earliest delegation to avoid querying a wide range
+                earliest_event = min(events, key=lambda event: event.timestamp)
```

Comment:
Can't you do that directly from the filter above? Use the `order_by` argument and make sure it returns events order by ascending timestamp? So then getting `events[0]` is the `earliest_event` and as such you don't need this at all.